src/streamlink/plugins/youtube.py

```diff
@@ -354,44 +331,37 @@ def _get_streams(self):
             log.error("Could not get video info")
             return
 
-        if info.get("livestream") == '1' or info.get("live_playback") == '1' \
-                or info.get("player_response", {}).get("videoDetails", {}).get("isLive"):
+        if info.get("player_response", {}).get("videoDetails", {}).get("isLiveContent"):
             log.debug("This video is live.")
             is_live = True
 
-        formats = info.get("fmt_list")
         streams = {}
-        protected = False
-        for stream_info in info.get("url_encoded_fmt_stream_map", []):
-            if stream_info.get("s"):
-                protected = True
-                continue
+        log.debug("{0}".format(info))
+        if (info.get("player_response", {}).get("streamingData", {}).get("adaptiveFormats", [])[0].get("cipher")
+           or info.get("player_response", {}).get("streamingData", {}).get("formats", [])[0].get("cipher")):
+            log.debug("This video is protected.")
```

Comment:
**won't work**
`[plugin.youtube][debug] This video is protected.`

```
$ streamlink --plugin-dirs gg/ https://www.youtube.com/watch?v=24_dtEGEYCI -l info
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/watch?v=24_dtEGEYCI
error: This plugin does not support protected videos, try youtube-dl instead
```

current version would work

```
$ streamlink https://www.youtube.com/watch?v=24_dtEGEYCI -l info
[cli][info] Found matching plugin youtube for URL https://www.youtube.com/watch?v=24_dtEGEYCI
[cli][info] Available streams: 144p (worst), 240p, 360p, 480p, 720p, 1080p (best)
[cli][info] Opening stream: 1080p (hls)
[cli][info] Starting player: /usr/bin/mpv
```

Reply:
Urls of your stream are actually ciphered (see [24_dtEGEYCI.json](https://gist.github.com/brouxco/68f02161fd7f8dc5779ac173b14f1b41))
It is not yet supported in streamlink AFAIK

Reply:
I changed it (see 00576f98)